### PR TITLE
Introducing Configuration management through Config file

### PR DESCRIPTION
**Pull request by yogpan01**
This commit includes
 - Support for configuring various build time parameters throuh user supplied config file
  that can be passed at compile time through Makefile system which will work for
  non mbed-OS and non yotta build system as well.

### CODE REVIEW

**Comment by ciarmcom** [comment]
Automatic CI verification build not done, please verify manually.

**Comment by yogpan01** [comment]
@anttiylitokola Please check sn_config.h has all the config flag defined which are supported through yotta config and mbed-cli config.

File: nsdl-c/sn_config.h [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+ */
+#undef SN_COAP_DUPLICATION_MAX_MSGS_COUNT   /* 0 */
+
+/**
+ * \def SN_COAP_MAX_BLOCKWISE_PAYLOAD_SIZE
+ *
+ * \brief For Message blockwising
+ * Init value for the maximum payload size to be sent and received at one blockwise message
+ * Setting of this value to 0 will disable this feature, and also reduce use of ROM memory
+ * Note: Current Coap implementation supports Blockwise transfers specification version draft-ietf-core-block-03
```
Comment (anttiylitokola) [inline nsdl-c/sn_config.h]: This NOTE line can be removed. We are supporting latest spec version.

File: nsdl-c/sn_config.h [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+*  parameters.
+*/
+
+/**
+ * \def SN_COAP_DUPLICATION_MAX_MSGS_COUNT
+ * \brief For Message duplication detection
+ * Init value for the maximum count of messages to be stored for duplication detection
+ * Setting of this value to 0 will disable duplication check, also reduce use of ROM memory
+ */
+#undef SN_COAP_DUPLICATION_MAX_MSGS_COUNT   /* 0 */
```
Comment (anttiylitokola) [inline nsdl-c/sn_config.h]: Default for this has changed to 1

File: nsdl-c/sn_config.h [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+ * Setting this to 0 disables this feature.
+ * By default, this feature is disabled.
+ */
+#undef SN_COAP_RESENDING_QUEUE_SIZE_BYTES   /* 0  */ // Default re-sending queue size - defines size of the re-sending buffer. Setting this to 0 disables feature
+
+/**
+ * \def SN_COAP_MAX_INCOMING_MESSAGE_SIZE
+ *
+ * \brief Sets the maximum size (in bytes) that
+ * mbed Client will alloe to be handle while
```
Comment (anttiylitokola) [inline nsdl-c/sn_config.h]: small type "alloe"

**Comment by anttiylitokola** [comment]
Only small comment changes, otherwise LGTM

**Comment by yogpan01** [comment]
rebuild

**Comment by yogpan01** [comment]
rebuild

**Comment by yogpan01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by yogpan01** [comment]
rebuild

**Comment by yogpan01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by teetak01** [comment]
rebuild

**Comment by yogpan01** [comment]
rebuild

**Pull request merged**